libvast/vast/row_major_matrix_table_slice_builder.hpp

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include "vast/data.hpp"
+#include "vast/table_slice.hpp"
+#include "vast/table_slice_builder.hpp"
+
+namespace vast {
+
+class row_major_matrix_table_slice_builder final : public table_slice_builder {
+public:
+  // -- member types -----------------------------------------------------------
+
+  using super = table_slice_builder;
+
+  row_major_matrix_table_slice_builder(record_type layout);
+
+  virtual ~row_major_matrix_table_slice_builder();
+
+  // -- factory functions ------------------------------------------------------
+
+  /// @returns a table slice builder instance.
+  static table_slice_builder_ptr make(record_type layout);
+
+  /// @returns a default-constructed table slice instance.
+  static table_slice_ptr make_slice(record_type layout, table_slice::size_type);
```

Comment:
Would you mind documenting the factory function parameters and return values? It seems the second parameter here is `rows`, and it would be good to convey the behavior, especially what this means with respect to memory allocations.